web/src/settings_components.ts

```diff
@@ -382,7 +382,8 @@ function read_select_field_data_from_form(
     $profile_field_form: JQuery,
     old_field_data: FieldData,
 ): FieldData {
-    const field_data: FieldData = {};
+    type FieldDataMap = Map<keyof FieldData, {text: string; order: string}>;
+    const field_data: FieldDataMap = new Map();
```

Comment:
I've only modified field_data of this function to use a Map and not other functions. Otherwise we'll have to convert the Map to an object at some point in `read_field_data_from_form`, so might as well convert it back in this function itself when this is the only function where a map is needed